src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs

```diff
@@ -971,7 +971,13 @@ private void StartStream()
                     throw new Http2StreamErrorException(_currentHeadersStream.StreamId, CoreStrings.Http2ErrorMissingMandatoryPseudoHeaderFields, Http2ErrorCode.PROTOCOL_ERROR);
                 }
 
-                if (_clientActiveStreamCount > _serverSettings.MaxConcurrentStreams)
+                if (_clientActiveStreamCount == _serverSettings.MaxConcurrentStreams)
+                {
+                    // Provide feedback in server logs that the client hit the number of maximum concurrent streams,
+                    // and that the client is likely waiting for existing streams to be completed before it can continue.
+                    Log.Http2ReachedMaxConcurrentStreams(_context.ConnectionId);
```

Comment:
This could be pretty noisy if a client has tons of pending requests and always opens a new stream as soon as it's able. What do you think about limiting this log to at most once a connection?

Reply:
I'd also be OK with keeping it as is and making it Trace-level.

Reply:
Changed to trace.